### PR TITLE
Add credit request for Turing BinderHub

### PR DESCRIPTION
**Pull request by sgibson91**
### Summary

This PR adds the credit request we submitted to fund the Turing mybinder federation cluster to the project management folder.

### What should a reviewer concentrate their feedback on?

- [ ] Everything looks ok?

### Acknowledging contributors

- [ ] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file.
- [x] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file:

### CODE REVIEW

File: project_management/turing_binderhub_azure_credit_request_Oct2019.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+Suggested name: turing-mybinder
+
+## Credit Request
+
+$10,000
+
+## Start/End dates for allocation
+
+Begin: 1st October 2019
+End: 30th September 2020
```
Comment (malvikasharan) [inline project_management/turing_binderhub_azure_credit_request_Oct2019.md]: Is this valid for 1 year only? Why is the allocation starting from 2019? Apologies if you have clarified that somewhere already.

Comment (sgibson91) [reply project_management/turing_binderhub_azure_credit_request_Oct2019.md]: Yes it's for one year, aligning with the financial/academic year. It's already been submitted/approved, @KirstieJane just asked that I added it to the repo so it's more findable than the hackmd it was developed in.

File: project_management/turing_binderhub_azure_credit_request_Oct2019.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+Binder is a community within Project Jupyter which received the ACM Software System Award in 2017 for language agnostic tools which have become "a de facto standard for data analysis in research, education, journalism, and industry" (https://awards.acm.org/award_winners/perez_9039634). Binder is a globally established but still growing project. The Turing Institute will be joining the team when the technology is tried and tested in production _and_ ready to be scaled up to support growing demand for transparent, reproducible, reusable research. Joining the BinderHub Federation promotes all three of the Institute's goals: advancing world-class research on real-world problems in academic, government and industry settings, providing interactive training materials for the next generation of data science leaders, and leading the public conversation on modular, interoperable, community owned and run infrastructure. Our shared goal is to democratise access to artificial intelligence for all.
```
Comment (malvikasharan) [inline project_management/turing_binderhub_azure_credit_request_Oct2019.md]:
```suggestion
Binder is a community within Project Jupyter which received the [ACM Software System Award](https://en.wikipedia.org/wiki/ACM_Software_System_Award) in 2017 for language-agnostic tools which have become "a de facto standard for data analysis in research, education, journalism, and industry" (https://awards.acm.org/award_winners/perez_9039634). 
Binder is a globally established but still growing project. [The Alan Turing Institute](https://www.turing.ac.uk/) will be joining the team when the technology is tried and tested in production _and_ ready to be scaled up to support growing demand for transparent, reproducible, reusable research. 
Joining the BinderHub Federation promotes all three of the Institute's goals: advancing world-class research on real-world problems in academic, government and industry settings, providing interactive training materials for the next generation of data science leaders, and leading the public conversation on modular, interoperable, community-owned and run infrastructure. Our shared goal is to democratise access to artificial intelligence for all.
```

File: project_management/turing_binderhub_azure_credit_request_Oct2019.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+## The Computational Requirements
+
+* A managed Kubernetes cluster of 3 Standard D4s v3 virtual machines. These VMs have 4 vCPUs and 16 GB RAM. This is roughly half the size of the VMs in the GKE cluster and similar in size to those in the OVH cluster. Based on GKE statistics (using "maximum" pods at https://grafana.mybinder.org/d/nDQPwi7mk/node-activity?panelId=26&fullscreen&orgId=1&from=1565856051535&to=1565873478287&var-cluster=prometheus), we'd expect this cluster to support ~30 users per node at full capacity, subject to RAM limitations.
```
Comment (malvikasharan) [inline project_management/turing_binderhub_azure_credit_request_Oct2019.md]:
```suggestion
* A managed Kubernetes cluster of 3 Standard D4s v3 virtual machines. These VMs have 4 vCPUs and 16 GB RAM. This is roughly half the size of the VMs in the GKE cluster and similar in size to those in the OVH cluster. Based on GKE statistics (using "maximum" pods at [grafana.mybinder.org cluster](https://grafana.mybinder.org/d/nDQPwi7mk/node-activity?panelId=26&fullscreen&orgId=1&from=1565856051535&to=1565873478287&var-cluster=prometheus)), we'd expect this cluster to support ~30 users per node at full capacity, subject to RAM limitations.
```

File: project_management/turing_binderhub_azure_credit_request_Oct2019.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+* An Azure Container Registry with a Premium SKU. The BinderHub builds Docker images and requires access to a container registry for storage. The Premium SKU includes 500 GB of storage in the cost estimate.
```
Comment (malvikasharan) [inline project_management/turing_binderhub_azure_credit_request_Oct2019.md]:
```suggestion
* An Azure Container Registry with a Premium service tier, i.e. SKUs. The BinderHub builds Docker images and requires access to a container registry for storage. The Premium SKU includes 500 GB of storage in the cost estimate.
```

File: project_management/turing_binderhub_azure_credit_request_Oct2019.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+* A B1S "burstable" virtual machine to run automated maintenance from. In particular, the container registry is expected to rapidly grow in size as more users are directed to the Azure cluster. The GKE registry usually reaches 50TB and the OVH registry reaches 2TB.
+
+Sarah Gibson, Turing REG team member, has developed an automated script (https://github.com/HelmUpgradeBot/DockerCleanUp) that checks the size of the registry and deletes old (last built more than a customisable number of days ago) or large images since Azure charge $0.003 per GB per day over the included 500 GB. By automatically running this script on an economic VM, we can reduce the likelihood of spending too many credits on the run-away growth of the container registry. We aim to match the 2TB maximum size of the OVH registry.
+
+## List of people who require access
+
+* Sarah Gibson: sgibson@turing.ac.uk, REG data scientist and mybinder.org operator.
```
Comment (malvikasharan) [inline project_management/turing_binderhub_azure_credit_request_Oct2019.md]:
```suggestion
* Sarah Gibson (email: [sgibson@turing.ac.uk](mailto:sgibson@turing.ac.uk)), REG data scientist and mybinder.org operator.
```

File: project_management/turing_binderhub_azure_credit_request_Oct2019.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+* Tim Head: betatim@gmail.com, Core member of mybinder.org maintenance and operating team.
```
Comment (malvikasharan) [inline project_management/turing_binderhub_azure_credit_request_Oct2019.md]:
```suggestion
* Tim Head (email: [betatim@gmail.com](mailto:betatim@gmail.com)), Core member of mybinder.org maintenance and operating team.
```

File: project_management/turing_binderhub_azure_credit_request_Oct2019.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+Our justification for having an external collaborator on this project is to ensure that Sarah is not the sole failure point for maintaining the cluster. Tim has extensive experience developing, maintaining and debugging BinderHub clusters and is an expert in this domain. His collaboration will also make it easier to disseminate knowledge about deploying and maintaining BinderHubs on Microsoft's Azure infrastructure into the wider BinderHub community.
```
Comment (malvikasharan) [inline project_management/turing_binderhub_azure_credit_request_Oct2019.md]:
```suggestion
An external collaborator on this project will ensure that Sarah is not solely maintaining the cluster. Tim has extensive experience developing, maintaining and debugging BinderHub clusters and is an expert in this domain. His collaboration will also make it easier to disseminate knowledge about deploying and maintaining BinderHubs on Microsoft's Azure infrastructure into the wider BinderHub community.
```

File: project_management/turing_binderhub_azure_credit_request_Oct2019.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+Collaborators who do not need access, but who will support Sarah and Tim in Turing's participation in the BinderHub community are Kirstie Whitaker (Turing Research Fellow) and Chris Holdgraf (Berkeley Institute for Data Science fellow, and core Project Jupyter team member).
```
Comment (malvikasharan) [inline project_management/turing_binderhub_azure_credit_request_Oct2019.md]: DO you wanna include email or contact info for Kirstie and Chris?

File: project_management/turing_binderhub_azure_credit_request_Oct2019.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+Resource | Number | Cost ($ per day) | Cost ($ per month) | Cost ($ per year) | Total Cost ($ per year) (Number * Cost per year)
+| :--- | ---: | ---: | ---: | ---: | ---: |
+Container Registry (Premium SKU) | 1 | 1.667 | | 608.872 | 608.872
+Extra Storage (up to 2TB, first 500 GB incl.) | 1 | 4.500 | | 1643.625 | 1643.625
+Standard D4s v3 VM | 3 | | 175.50 | 2106.00 | 6318.00
+B1S VM | 1 | | 8.76 | 105.12 | 105.12
+| | | | | |
+| | | | | **Total Cost ($)** | 8675.62
+
+The $10,000 we have asked for includes an approximate 10% buffer to support testing and development requirements. While it is not needed for standard running of the Turing BinderHub, we will use it to experiment with an auto-scaling Kubernetes cluster that will make the BinderHub more cost effective in the long run.
```
Comment (malvikasharan) [inline project_management/turing_binderhub_azure_credit_request_Oct2019.md]: Maybe include the 10% in the table rather than explaining it afterwards.

**Comment by sgibson91** [comment]
Hi @malvikasharan so sorry for the confusion 🙀This proposal was submitted and approved last September and the Turing instance of mybinder.org is now up and running - I only recently got around to adding it to the repo. I think we should merge this as is since that's the state the proposal was accepted in. Thank you for the comments though 💖

**Comment by sgibson91** [comment]
I will commit the suggestions that put new sentences on new lines - thank you!

**Pull request merged**